java/ql/src/experimental/CWE-532/SensitiveInfoLog.ql

```diff
@@ -0,0 +1,82 @@
+/**
+ * @id java/sensitiveinfo-in-logfile
+ * @name Insertion of sensitive information into log files
+ * @description Writing sensitive information to log files can give valuable guidance to an attacker or expose sensitive user information.
+ * @kind problem
+ * @tags security
+ *       external/cwe-532
+ */
+
+import java
+import semmle.code.java.dataflow.TaintTracking
+import DataFlow
+import PathGraph
+
+/** Class of popular logging utilities **/
+class LoggerType extends RefType {
+    LoggerType() {
+        this.hasQualifiedName("org.apache.log4j", "Category") or //Log4J
+        this.hasQualifiedName("org.slf4j", "Logger")  //SLF4j and Gradle Logging
+    }
+}
+
+/** Concatenated string with a variable that keeps sensitive information judging by its name **/
+class CredentialExpr extends Expr {
+    CredentialExpr() {
+        exists (Variable v | this.(AddExpr).getAnOperand() = v.getAnAccess() | v.getName().regexpMatch(getACredentialRegex()))
+    }
+}
+
+/** Source in concatenated string or variable itself  **/
+class CredentialSource extends DataFlow::ExprNode {
+    CredentialSource() {
+        exists (
+            Variable v | this.asExpr() = v.getAnAccess() | v.getName().regexpMatch(getACredentialRegex())
+        ) or
+        exists (
+            this.asExpr().(AddExpr).getAnOperand().(CredentialExpr)  
+        ) or
```

Comment:
The pattern `exists(<expr>.(Type))` can be written simpler as `<expr> instanceof Type`.

Reply:
Also, instead of adding a few cases for string concatenation here in the source, it's much better to override `predicate isAdditionalFlowStep(Node node1, Node node2)` in the configuration to let the dataflow use string concatenation as a general step (`TaintTracking::Configuration` does this out of the box, but that also includes other taint steps, which likely are undesirable, so it makes sense to keep this as a `DataFlow::Configuration` and just add the relevant string concatenation steps).

Reply:
Hi @aschackmull ,

Thanks for reviewing this PR. Yes, I've tested this query on some open-source projects that have most forks, and here are several of them and  what I learned from looking through the results:
1. https://github.com/Olybrius/JavaEE-EventManager
<img width="909" alt="olybrius" src="https://user-images.githubusercontent.com/37377780/81626709-7727df00-93ca-11ea-886c-adde579ce298.png">

2. https://github.com/Atmosphere/atmosphere
<img width="901" alt="atmosphere" src="https://user-images.githubusercontent.com/37377780/81626715-7f801a00-93ca-11ea-9701-4bab1093bc85.png">

 3. https://github.com/prashanth-sams/re-runFailedTests
<img width="926" alt="prashanth-sams" src="https://user-images.githubusercontent.com/37377780/81626728-89098200-93ca-11ea-9645-06270ee2f1e0.png">

I also noticed some projects do have this issue but they cannot be compiled by the tool thus no database was created unless I clone a copy then make it compilable. E.g.
4. https://github.com/piergiorgiocremonese/roncovillage/
<img width="1079" alt="piergiorgiocremonese" src="https://user-images.githubusercontent.com/37377780/81627546-6c6e4980-93cc-11ea-8bd5-c4fc845155f5.png">

I know logging sensitive information is a big and common issue with application development so I believe it's an interesting topic for CodeQL queries. I will update the query with your suggestions, but please bear with me as I'm still learning CodeQL.

@luchua-bc